### PR TITLE
fix(ONYX-826): fix performance after alert deletion

### PR DESCRIPTION
**Pull request by dariakoko**
The type of this PR is: **TYPE**

This PR solves [ONYX-826]

### Description

Select another alert after deletion

| Before | After |
| --- | --- |
| <video src="https://github.com/artsy/force/assets/36167539/050e4f72-142e-4e4f-8bca-e90078bca32d"> | <video src="https://github.com/artsy/force/assets/36167539/9d1c72aa-ce83-4ddc-9d5d-2ff8d377c662"> |

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ONYX-826]: https://artsyproduct.atlassian.net/browse/ONYX-826?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dariakoko** [comment]
The performance doesn't look very smooth. Let's explore the options and address it in another PR

File: src/Apps/Settings/Routes/SavedSearchAlerts/SavedSearchAlertsApp.tsx [hunk 1]
```diff
@@ -138,12 +138,33 @@ export const SavedSearchAlertsApp: React.FC<SavedSearchAlertsAppProps> = ({
...
+    isMobile ? refresh() : closeEditFormAndRefetch()
     closeDeleteModal()
 
     sendToast({
       message: "Your Alert has been deleted.",
     })
+
+    if (isMobile) return
+
+    if (editAlertEntity?.id === alerts[0].internalID) {
```
Comment (MounirDhahri) [inline src/Apps/Settings/Routes/SavedSearchAlerts/SavedSearchAlertsApp.tsx:150]: A comment here would be great to make sure it's clear what this does

File: src/Apps/Settings/Routes/SavedSearchAlerts/SavedSearchAlertsApp.tsx [hunk 2]
```diff
@@ -138,12 +138,33 @@ export const SavedSearchAlertsApp: React.FC<SavedSearchAlertsAppProps> = ({
...
+      setEditAlertEntity({
+        id: alerts[1].internalID,
```
Comment (MounirDhahri) [inline src/Apps/Settings/Routes/SavedSearchAlerts/SavedSearchAlertsApp.tsx:152]: are we always sure we have `alerts[1]`, wondering about the case where we have only one alert

**Review by MounirDhahri — COMMENTED** [review]
A small comment about the existence of a second alert, otherwise looks good to me

File: src/Apps/Settings/Routes/SavedSearchAlerts/SavedSearchAlertsApp.tsx (hunk 2 above)
Comment (dariakoko) [reply src/Apps/Settings/Routes/SavedSearchAlerts/SavedSearchAlertsApp.tsx:155]: How did i miss that 🥴  
Corrected the condition, thank you

File: src/Apps/Settings/Routes/SavedSearchAlerts/SavedSearchAlertsApp.tsx [hunk 3]
```diff
@@ -147,21 +147,28 @@ export const SavedSearchAlertsApp: React.FC<SavedSearchAlertsAppProps> = ({
 
     if (isMobile) return
 
-    if (editAlertEntity?.id === alerts[0].internalID) {
+    // the right collumn of the screen should always display alert's edit form
+    // this is why when we delete and alert we update the editAlertEntity
+    // to contain the details of the first alert on the alerts' list
+    if (editAlertEntity?.id === alerts[0].internalID && alerts.length > 1) {
```
Comment (MounirDhahri) [inline src/Apps/Settings/Routes/SavedSearchAlerts/SavedSearchAlertsApp.tsx:153]: Awesome, thanks for the adjustment

File: src/Apps/Settings/Routes/SavedSearchAlerts/SavedSearchAlertsApp.tsx [hunk 4]
```diff
@@ -147,21 +147,28 @@ export const SavedSearchAlertsApp: React.FC<SavedSearchAlertsAppProps> = ({
...
       setEditAlertEntity({
         id: alerts[1].internalID,
         name: alerts[1].settings.name ?? "",
         artistIds: alerts[1]?.artistIDs as string[],
       })
       silentPush(`/settings/alerts/${alerts[1].internalID}/edit`)
-    } else if (editAlertEntity?.id !== alerts[0].internalID) {
+    } else if (
+      editAlertEntity?.id !== alerts[0].internalID &&
+      alerts.length > 1
```
Comment (MounirDhahri) [inline src/Apps/Settings/Routes/SavedSearchAlerts/SavedSearchAlertsApp.tsx:162]: is this check needed? i think the first one is enough

**Review by MounirDhahri — COMMENTED** [review]
LGTM with just a small question

**Review by MounirDhahri — APPROVED** [review]

**Pull request merged**